context/authContext.tsx

```diff
@@ -0,0 +1,37 @@
+import { createContext, useContext, useEffect, useState } from "react";
+
+export const AuthContext = createContext({});
+
+export type AuthUser = {
+  name: string;
+  email: string;
+  photo: string;
+  uid: string;
+};
```

Comment:
Although a type works here, this would be a great place to use an interface instead as AuthUser is an object,

Reply:
Thanks, @emmalearnscode for the feedback.  Curious to know how would `interface` help over `type` here.

Reply:
Here they are interchangeable but in most projects you will see interfaces used instead of types for objects as interfaces are more flexible, they can be extended and built upon whereas types cannot.

Reply:
Understood. Thanks, @emmalearnscode